### PR TITLE
Mute connection closed error from websocket

### LINKED ISSUE #3758: connection handler failed error in console when MAX_RUNTIME is hit
**Describe the bug**
`connection handler failed` message is produced in user prompt when all jobs fail due to reaching `MAX_RUNTIME`. Tested when only some jobs fail and in that scenario connections are handled correctly.

**To reproduce**
Steps to reproduce the behaviour:
1. `pip install ert`
2. Set `MAX_RUNTIME` to low number
3. Modify `poly.py` to include sleep higher than `MAX_RUNTIME` 
4. Run experiment (IES/Smoother/ESMDA/Test)

**Expected behaviour**
Handle websocket connection failure properly, does not show `connection handler failed`.

**Log**

<details>
  <summary>ert log</summary>
  
```
2022-08-15 08:19:47,251 - websockets.client - Thread-2 - DEBUG - > PING 3e 59 62 bd [binary, 4 bytes]
2022-08-15 08:19:47,254 - websockets.client - Thread-2 - DEBUG - < PONG 3e 59 62 bd [binary, 4 bytes]
2022-08-15 08:19:47,254 - websockets.client - Thread-2 - DEBUG - % received keepalive pong
2022-08-15 08:19:53,955 - websockets.server - ert_ee_run_server - ERROR - connection handler failed
Traceback (most recent call last):
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 944, in transfer_data
    message = await self.read_message()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1013, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1088, in read_data_frame
    frame = await self.read_frame(max_size)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1143, in read_frame
    frame = await Frame.read(
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/framing.py", line 70, in read
    data = await reader(2)
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/asyncio/streams.py", line 723, in readexactly
    await self._wait_for_data('readexactly')
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/asyncio/streams.py", line 517, in _wait_for_data
    await self._waiter
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/asyncio/selector_events.py", line 848, in _read_ready__data_received
    data = self._sock.recv(self.max_size)
ConnectionResetError: [Errno 104] Connection reset by peer

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/server.py", line 232, in handler
    await self.ws_handler(self)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/server.py", line 1144, in _ws_handler
    return await cast(
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/ensemble_evaluator/evaluator.py", line 246, in connection_handler
    await self.handle_dispatch(websocket, path)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/ensemble_evaluator/evaluator.py", line 214, in handle_dispatch
    async for msg in websocket:
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 481, in __aiter__
    yield await self.recv()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 552, in recv
    await self.ensure_open()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 920, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: no close frame received or sent
2022-08-15 08:19:54,019 - websockets.server - ert_ee_run_server - ERROR - connection handler failed
Traceback (most recent call last):
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 944, in transfer_data
    message = await self.read_message()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1013, in read_message
    frame = await self.read_data_frame(max_size=self.max_size)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1088, in read_data_frame
    frame = await self.read_frame(max_size)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 1143, in read_frame
    frame = await Frame.read(
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/framing.py", line 70, in read
    data = await reader(2)
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/asyncio/streams.py", line 721, in readexactly
    raise exceptions.IncompleteReadError(incomplete, n)
asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of 2 expected bytes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/server.py", line 232, in handler
    await self.ws_handler(self)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/server.py", line 1144, in _ws_handler
    return await cast(
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/ensemble_evaluator/evaluator.py", line 246, in connection_handler
    await self.handle_dispatch(websocket, path)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/ensemble_evaluator/evaluator.py", line 214, in handle_dispatch
    async for msg in websocket:
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 481, in __aiter__
    yield await self.recv()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 552, in recv
    await self.ensure_open()
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/websockets/legacy/protocol.py", line 920, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: no close frame received or sent
2022-08-15 08:20:07,002 - websockets.client - Thread-1 - DEBUG - < PING ff de 53 21 [binary, 4 bytes]
2022-08-15 08:20:07,003 - websockets.client - Thread-1 - DEBUG - > PONG ff de 53 21 [binary, 4 bytes]
2022-08-15 08:20:07,007 - websockets.client - Thread-1 - DEBUG - % sending keepalive ping
2022-08-15 08:20:07,007 - websockets.client - Thread-1 - DEBUG - > PING c7 4c cb 63 [binary, 4 bytes]
2022-08-15 08:20:07,010 - websockets.client - Thread-1 - DEBUG - < PONG c7 4c cb 63 [binary, 4 bytes]
2022-08-15 08:20:07,011 - websockets.client - Thread-1 - DEBUG - % received keepalive pong
2022-08-15 08:20:07,251 - websockets.client - Thread-2 - DEBUG - < PING 'މ4\x08' [text, 4 bytes]
2022-08-15 08:20:07,251 - websockets.client - Thread-2 - DEBUG - > PONG 'މ4\x08' [text, 4 bytes]
2022-08-15 08:20:07,256 - websockets.client - Thread-2 - DEBUG - % sending keepalive ping
2022-08-15 08:20:07,257 - websockets.client - Thread-2 - DEBUG - > PING c0 2e bd 24 [binary, 4 bytes]
2022-08-15 08:20:07,259 - websockets.client - Thread-2 - DEBUG - < PONG c0 2e bd 24 [binary, 4 bytes]
2022-08-15 08:20:07,260 - websockets.client - Thread-2 - DEBUG - % received keepalive pong
2022-08-15 08:20:08,821 - websockets.client - Thread-2 - DEBUG - < TEXT '{"specversion": "1.0", "id": "17adf67f-5411-46b...06:20:08.819654+00:00"}' [183 bytes]
2022-08-15 08:20:08,825 - websockets.client - Thread-1 - DEBUG - < TEXT '{"specversion": "1.0", "id": "17adf67f-5411-46b...06:20:08.819654+00:00"}' [183 bytes]
2022-08-15 08:20:08,826 - websockets.client - Thread-2 - DEBUG - = connection is CLOSING
2022-08-15 08:20:08,827 - websockets.client - Thread-1 - DEBUG - < CLOSE 1001 (going away) [2 bytes]
2022-08-15 08:20:08,827 - websockets.client - Thread-2 - DEBUG - > CLOSE 1000 (OK) [2 bytes]
2022-08-15 08:20:08,828 - websockets.client - Thread-1 - DEBUG - = connection is CLOSING
2022-08-15 08:20:08,830 - websockets.client - Thread-2 - DEBUG - < CLOSE 1001 (going away) [2 bytes]
2022-08-15 08:20:08,830 - websockets.client - Thread-1 - DEBUG - > CLOSE 1001 (going away) [2 bytes]
2022-08-15 08:20:08,833 - websockets.client - Thread-2 - DEBUG - = connection is CLOSED
2022-08-15 08:20:08,833 - websockets.client - Thread-2 - DEBUG - x closing TCP connection
2022-08-15 08:20:08,840 - websockets.client - Thread-1 - DEBUG - = connection is CLOSED
2022-08-15 08:20:08,841 - websockets.client - Thread-1 - DEBUG - x closing TCP connection
2022-08-15 08:20:13,853 - root - MainThread - DEBUG - Resetting environment variable ERT_SITE_CONFIG
2022-08-15 08:20:13,854 - root - MainThread - DEBUG - Deleting temporary directory for site-config
2022-08-15 08:20:13,855 - res.enkf - DrainerThread - DEBUG - Calling decref on: storage/default . Refcount after decref:2
2022-08-15 08:20:13,856 - ert_shared.main - MainThread - ERROR - Simulation failed! All realizations failed!
connection handler failed
connection handler failed
Traceback (most recent call last):
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/main.py", line 499, in main
    args.func(args)
  File "/prog/res/komodo/2022.08.01-py38-rhel7/root/lib64/python3.8/site-packages/ert_shared/cli/main.py", line 95, in run_cli
    raise ErtCliError(model.getFailMessage())
ert_shared.cli.main.ErtCliError: Simulation failed! All realizations failed!
connection handler failed
connection handler failed
```

</details>

**Environment**
 - OS: [RHEL7]
 - ERT/Komodo release: [2022.08.01]
 - Python version: 3.8
 - Remote/HPC execution involved: [yes|no] (tested both - same behaviour)

### PR DESCRIPTION
**Pull request by berland**
Whenever dispatchers die (killed by ERT f.ex or for other reasons)
the websocket is not properly closed and the exception is not caught. That leads to a situation that needs a timeout to resolve (20 seconds) (it is waiting for dispatchers to disconnect, even when there is none)

**Issue**
Resolves #3758 

**Approach**
Catch and log exception. Then carry on.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: test-data/local/poly_example/poly_eval.py [hunk 1]
```diff
@@ -9,6 +10,7 @@ def _load_coeffs(filename):
 
 
 def _evaluate(coeffs, x):
+    time.sleep(300)
```
Comment (xjules) [inline test-data/local/poly_example/poly_eval.py:13]: I guess this is left-over from testing

File: src/ert/shared/ensemble_evaluator/evaluator.py [hunk 2]
```diff
@@ -209,32 +209,38 @@ async def handle_dispatch(self, websocket, path):
...
+                        )
+                        await websocket.close(code=1011, reason=f"failed handling {event}")
+                        return
+
+                    if event["type"] in [
+                        identifiers.EVTYPE_ENSEMBLE_STOPPED,
+                        identifiers.EVTYPE_ENSEMBLE_FAILED,
+                    ]:
+                        return
+            except ConnectionClosedError as connection_error:
```
Comment (xjules) [inline src/ert/shared/ensemble_evaluator/evaluator.py:239]: Still need to think about this :)

File: src/ert/shared/ensemble_evaluator/evaluator.py [hunk 3]
```diff
@@ -209,32 +209,45 @@ async def handle_dispatch(self, websocket, path):
...
+                        logger.info(
+                            "Got event from evaluator "
+                            f"{self._get_ens_id(event['source'])} "
+                            f"with source {event['source']}, "
+                            f"ignoring since I am {self.ensemble.id_}"
+                        )
+                        continue
+                    try:
+                        await self._dispatcher.handle_event(event)
+                    except BaseException as ex:  # pylint: disable=broad-except
```
Comment (xjules) [inline src/ert/shared/ensemble_evaluator/evaluator.py:228]: This is not part of this PR, but handling `BaseException` does not look right.

File: src/ert/shared/ensemble_evaluator/evaluator.py [hunk 4]
```diff
@@ -209,32 +209,45 @@ async def handle_dispatch(self, websocket, path):
...
+                        identifiers.EVTYPE_ENSEMBLE_STOPPED,
+                        identifiers.EVTYPE_ENSEMBLE_FAILED,
+                    ]:
+                        return
+            except ConnectionClosedError as connection_error:
+                # Dispatchers my close the connection apruptly in the case of
+                #  * flaky network (then the dispatcher will try to reconnect)
+                #  * job being killed due to MAX_RUNTIME
+                #  * job being killed by user
+                logger.debug(
```
Comment (xjules) [inline src/ert/shared/ensemble_evaluator/evaluator.py:248]: `logger.error` instead?

File: src/ert/shared/ensemble_evaluator/evaluator.py (hunk 3 above)
Comment (berland) [reply src/ert/shared/ensemble_evaluator/evaluator.py:228]: Agree. At least  `handle_event`  may raise `asyncio.InvalidStateError` but then also a pretty general ` await f(event) ` is involved here where  `f`  is a step (?).

File: src/ert/shared/ensemble_evaluator/evaluator.py (hunk 4 above)
Comment (berland) [reply src/ert/shared/ensemble_evaluator/evaluator.py:248]: ok!

**Review by xjules — APPROVED** [review]
Bra job!m :rocket:
